app/src/main/java/org/schabi/newpipe/views/ScrollableTabLayout.java

```diff
@@ -0,0 +1,91 @@
+package org.schabi.newpipe.views;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewTreeObserver.OnGlobalLayoutListener;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.material.tabs.TabLayout;
+import com.google.android.material.tabs.TabLayout.Tab;
+
+/**
+ * A TabLayout that is scrollable when tabs exceed its width.
+ * Hides when there are less than 2 tabs.
+ */
+public class ScrollableTabLayout extends TabLayout {
+    private static final String TAG = ScrollableTabLayout.class.getSimpleName();
+
+    public ScrollableTabLayout(Context context) {
+        super(context);
+    }
+
+    public ScrollableTabLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public ScrollableTabLayout(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+
+        setTabMode(TabLayout.MODE_FIXED);
+        resetMode();
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+
+        resetMode();
+    }
+
+    @Override
+    public void addTab(@NonNull Tab tab, int position, boolean setSelected) {
+        super.addTab(tab, position, setSelected);
+
+        resetMode();
+    }
+
+    @Override
+    public void removeTabAt(int position) {
+        super.removeTabAt(position);
+
+        resetMode();
+    }
+
+    private void resetMode() {
+        if (getTabCount() < 2) {
+            setVisibility(View.GONE);
+            return;
+        } else {
+            setVisibility(View.VISIBLE);
+        }
+
+        if (getTabCount() == 0 || getTabAt(0).view == null) return;
+        setTabMode(TabLayout.MODE_FIXED);
+
+        int layoutWidth = getWidth();
+        int minimumWidth = ((View) getTabAt(0).view).getMinimumWidth();
+        if (minimumWidth * getTabCount() > layoutWidth) {
```

Comment:
Aren't you assuming here that the width of every tab is higher than or equal to the width of the first tab?

Take this example:
- there are two tabs of 500 and 300 pixels respectively: ----- ---
- the layout width is 900 pixels: --------

In this case the layout would be made scrollable (since 500*2 > 900), even though the views fit (500+300 <= 900).

I would remove the first layout width check completely, even though it could sometimes increase performance. What do you think? Another solution (that would solve this performance problem) would be to just keep track of the sum of the minimum size of every view by adding in `addTab()` and subtracting in `removeTabAt()`, but I don't know if the widths there would be correct.

Reply:
I checked implementation of `View` and `getWidth()` is constant time so I can remove this.

As for why I've done that - I assumed that `minWidth` of View wouldn't be changed by anything besides parent element (`TabLayout` in this case) so that every tab would have same value but that might not be correct. I would have used `getTabMinWidth()` but it's private. I could try reading it the same way it's done by `TabLayout`:
```java
requestedTabMinWidth =
        a.getDimensionPixelSize(R.styleable.TabLayout_tabMinWidth, INVALID_WIDTH);
```

I don't think it can be done in `addTab()` if we assume that width of some tab may later change.